### ISSUE TITLE
Cannot build for iOS with ionic cloud

### ISSUE THREAD

**Issue by dennis3001** [dpa99c/cordova-diagnostic-plugin#122]
I see this error when I try to build the my app with ionic:

Plugins/cordova.plugins.diagnostic/Diagnostic.m:13:9: fatal error: 'UserNotifications/UserNotifications.h' file not found
#import <UserNotifications/UserNotifications.h>
        ^
1 error generated

I have latest version 3.2.0

Any ideas?

**Comment by alepht** [comment]
Same here..

**Comment by dpa99c** [comment]
3.2.0 addresses deprecations made in iOS 10. I guess it's likely Ionic Cloud is not yet building against iOS 10, which adds the UserNotifications framework. My advice is build using v3.1 of the plugin until it does.

**Comment by mhfen** [comment]
Confirmed that is the issue @dpa99c . I changed my `package.json` to include `"cordova.plugins.diagnostic@3.1"` and the issue was resolved.

This will likely start to happen with many Cordova Plugins, you just seem to be the most responsible when it comes to updating your codebase for iOS releases.

There is a weird gray area right now, especially with plugins that used Swift, since there were so many breaking changes in Swift 3.0. Many ionic users will not be able to upgrade xcode until those issues are fixed in Swift plugins.

**Comment by dpa99c** [comment]
@mhfen Thanks for confirming my suspicion. There is always a gap between the latest platform release and the plugin authors catching up. I would hazard a guess that users of Phonegap Build will encounter the same issue that you have with Ionic Build until such time they both update their build environments.

I know exactly what you mean when it comes to Swift backward incompatibilities in XCode 8 - I have encountered it myself.  FYI I worked around it with plugins that use Swift 2.x syntax by opening the project in XCode and setting "Build Settings" > "Swift Compiler - Version" > "Use Legacy Swift Language Version" to "YES". Of course, if you use a cloud-based build environment, this is no help.
